AzureCalling/Controllers/AudioDeviceSelectionViewController.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+class AudioDeviceSelectionViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
+
+    // MARK: Properties
+
+    private var audioDeviceOptions: [BottomDrawerItem] = [BottomDrawerItem]()
+    var deviceTable: UITableView!
+
+    // MARK: UIViewController events
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.modalTransitionStyle = .crossDissolve
+
+        view.backgroundColor = UIColor(red: 0, green: 0, blue: 0, alpha: 0.8)
+        view.isOpaque = false
+
+        // tapping anywhere on the view is the same as tapping cancel
+        let tap = UITapGestureRecognizer(target: self, action: #selector(dismissSelf))
+        tap.cancelsTouchesInView = false
+        view.addGestureRecognizer(tap)
+
+        createAudioDeviceOptions()
+        createDeviceTable()
+    }
+
+    @objc func dismissSelf() {
+        dismiss(animated: true, completion: nil)
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        openDeviceTable()
+    }
+
+    func createDeviceTable() {
+        deviceTable = UITableView()
+        deviceTable.isScrollEnabled = false
+        deviceTable.layer.cornerRadius = 8
+        deviceTable.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(deviceTable)
+        let audioDeviceCell = UINib(nibName: "BottomDrawerCellView",
+                                      bundle: nil)
+        deviceTable.register(audioDeviceCell, forCellReuseIdentifier: "BottomDrawerCellView")
+        deviceTable.dataSource = self
+        deviceTable.delegate = self
+        deviceTable.reloadData()
+
+        let window = UIApplication.shared.windows[0]
+        let guide = view.safeAreaLayoutGuide
+        let bottomPadding = window.safeAreaInsets.bottom
+
+        var deviceTableConstraints = [
+            deviceTable.leadingAnchor.constraint(equalTo: guide.leadingAnchor),
+            deviceTable.trailingAnchor.constraint(equalTo: guide.trailingAnchor),
+            deviceTable.heightAnchor.constraint(equalToConstant: deviceTable.contentSize.height + bottomPadding)
```

Comment:
Found an edge case issue with the `heightAnchor` using a constant value while I'm working on the participant list. I know it's highly unlikely that we have large amount of audio device options to choose from, but the table doesn't display large list properly.

I have 2 suggestions on this:
1. If we can guarantee that the number of devices is small and still want the table to cover the full screen, then set `heightAnchor` with the argument `equalTo: guide.heightAnchor` instead of using `equalToConstant`. Otherwise, see below.
2. Use `topAnchor.constraint(equalTo: guide.centerYAnchor)` instead of `heightAnchor`, and set `isScrollEnabled` to true. This will set the table view to occupy half the screen and will have the same behavior as displaying the participant list.

Feel free to provide other suggestions if necessary.

Reply:
Since the Audio Device list is always static in this context, we can keep it the way it is without issue.  For participant list, we should probably use dynamic to content size up to a certain maximum size, and then make it scrollable, such as your 2nd suggestion.  So I agree it should be done as part of its refactoring to a common class and extension for participant list, but not for this particular ticket.

Reply:
For sure. We could do the refactoring separately if that's more appropriate.